elf/elf_file.cpp

```diff
@@ -260,6 +260,29 @@ void elf_file::read_sh(void) {
     }
 }
 
+// If there are holes between sections within segments, increase the section size to plug the hole
+// This is necessary to ensure the whole segment contains data defined in sections, otherwise you end up
+// signing/hashing/encrypting data that may not be written, as many tools write in sections not segments
+void elf_file::remove_sh_holes(void) {
+    for (int i=0; i+1 < sh_entries.size(); i++) {
+        auto sh0 = &(sh_entries[i]);
+        elf32_sh_entry sh1 = sh_entries[i+1];
+        if (
+            (sh0->type == SHT_PROGBITS && sh1.type == SHT_PROGBITS)
+            && (sh0->size && sh1.size)
```

Comment:
Is there a good reason for having `sh0` be a pointer, but `sh1` not be a pointer? IMHO `sh0->size && sh1.size` looks quite weird :wink:

Reply:
I did it because `sh0` can be modified but `sh1` doesn't need to be, so `sh0` has to be a pointer but `sh1` can just be a copy

Reply:
Fair enough.